File: data/EXP01/python/2406/2406.output_2.py
# @lc app=leetcode id=2406 lang=python3
class Solution:
    def minGroups(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[0])  # Sort intervals based on start values

        groups = []  # List to store the groups
        last_end = float('-inf')  # Keep track of the end value of the last added interval

        for interval in intervals:
            start, end = interval
            if start > last_end:
                groups.append([interval])
                last_end = end

            else:
                for group in groups:
                    if group[-1][1] >= start:
                        group.append(interval)
                        last_end = max(last_end, end)
                        break

        return len(groups)

File: data/EXP01/golang_no_system_var/2406/2406.output_2.go
func minGroups(intervals [][]int) int {
    sort.Slice(intervals, func(i, j int) bool {
        return intervals[i][0] < intervals[j][0]
    })

    groups := 1
    endTime := intervals[0][1]

    for i := 1; i < len(intervals); i++ {
        if endTime < intervals[i][0] {
            groups++
            endTime = intervals[i][1]
        } else {
            if endTime < intervals[i][1] {
                endTime = intervals[i][1]
            }
        }
    }

    return groups
}
